### ISSUE TITLE
dev-corpus2 breaks sentometrics

### ISSUE THREAD

**Issue by kbenoit** [quanteda/quanteda#1587]
From revdepcheck

# sentometrics

Version: 0.5.6

## Newly broken

*   checking examples ... ERROR
    ```
    ...
    Running examples in ‘sentometrics-Ex.R’ failed
    The error most likely occurred in:
    
    > ### Name: add_features
    > ### Title: Add feature columns to a (sento)corpus object
    > ### Aliases: add_features
    > 
    > ### ** Examples
    > 
    > data("usnews", package = "sentometrics")
    > 
    > set.seed(505)
    > 
    > # construct a corpus and add (a) feature(s) to it
    > corpus <- quanteda::corpus_sample(sento_corpus(corpusdf = usnews), 500)
    Warning: metacorpus argument is not used in sento_corpus()
    Warning in corp$tokens <- NULL : Coercing LHS to a list
    Error in corpus_sample.default(sento_corpus(corpusdf = usnews), 500) : 
      corpus_sample() only works on corpus objects.
    Calls: <Anonymous> -> corpus_sample.default
    Execution halted
    ```
*   checking tests ...
    ```
     ERROR
    Running the tests in ‘tests/testthat.R’ failed.
    Last 13 lines of output:
      ══ testthat results  ════════════════════════════════════════════════════════════════════════════════════
      OK: 15 SKIPPED: 0 FAILED: 10
      1.  Error: (unknown) (@test_aggregation.R#11) 
      2.  Error: (unknown) (@test_attribution.R#11) 
      3.  Failure: Corpus building works and fails when appropriate (@test_corpus_building.R#15) 
      4.  Error: Corpus building works and fails when appropriate (@test_corpus_building.R#17) 
      5.  Error: Conversion to sentocorpus from quanteda corpus (@test_corpus_building.R#26) 
      6.  Error: (unknown) (@test_corpus_building.R#35) 
      7.  Error: (unknown) (@test_measures_manipulation.R#11) 
      8.  Error: (unknown) (@test_methods_sentomeasures.R#11) 
      9.  Error: (unknown) (@test_model_specification.R#11) 
      10. Error: (unknown) (@test_sentiment_computation.R#21) 
      
      Error: testthat unit tests failed
      Execution halted
    ```

**Comment by kbenoit** [comment]
I did some checking, it's:

* in `sento_corpus()`, `corpus(..., metacorpus = ...)` is called:
    - https://github.com/sborms/sentometrics/blob/master/R/sentocorpus.R#L103
    - https://github.com/sborms/sentometrics/blob/master/R/sentocorpus.R#L124
    - https://github.com/sborms/sentometrics/blob/master/R/sentocorpus.R#L125 accesses a corpus internal using `$`

* These access the corpus `$documents` directly 😞: 
    - https://github.com/sborms/sentometrics/blob/master/R/sentocorpus.R#L131-L136
    - https://github.com/sborms/sentometrics/blob/master/R/sentocorpus.R#L173
    - https://github.com/sborms/sentometrics/blob/master/R/sentocorpus.R#L235

**Comment by kbenoit** [comment]
Fixed and passes for 1.4 and dev-corpus2, at least in my PR https://github.com/sborms/sentometrics/pull/5.
